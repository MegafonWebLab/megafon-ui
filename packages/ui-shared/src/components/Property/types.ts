export type Desc = {
    value: string[];
    isCollapsible?: boolean;
};

export type Item = {
    title?: string[];
    description?: Desc[];
    value?: string;
};
