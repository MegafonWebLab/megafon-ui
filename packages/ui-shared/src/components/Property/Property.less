@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-beta-property');

@{b} {
    display: flex;
    flex-direction: column;
    padding: 24px;
    border-top: 1px solid @spbSky2;

    @media @mobileSM {
        padding-left: 0;
        padding-right: 0;
    }

    @media @desktopMU {
        padding-top: 26px;
        padding-bottom: 26px;
    }

    &_border-bottom {
        border-bottom: 1px solid @spbSky2;
    }

    &__content {
        display: flex;
    }

    &__item {
        display: flex;

        &_multirow:first-child {
            margin-bottom: 10px;
        }
    }

    &__icon {
        width: 24px;
        height: 24px;
        overflow: hidden;
        margin-right: 9px;
        margin-left: 1px;
        display: inline-block;
        vertical-align: bottom;
    }

    &__badge {
        border-radius: 2px;
        font-size: 12px;
        padding: 2px 8px;
        color: @clearWhite;
        background: @mandarin;
    }

    &__inner {
        flex-grow: 1;
        padding-right: 20px;

        @media @mobileB {
            padding-right: 77px;
        }

        @media @mobileSM {
            padding-right: 8px;
        }
    }

    &__items-wrapper {
        flex-grow: 1;
    }

    &__value-wrapper {
        width: 47%;
        display: flex;
        justify-content: flex-end;
        align-items: flex-start;
        flex-shrink: 0;

        @media @desktopS {
            width: 35%;
        }

        @media @mobileB {
            width: 30%;
        }

        @media @mobileSM {
            width: 50%;
        }
    }

    &__value-wrapper_merged {
        border-left: 1px solid @spbSky2;
        align-items: center;
    }

    &__desc {
        margin-top: 8px;
    }

    &__item ~ &__item {
        margin-top: 16px;
    }

    &__value {
        max-width: 234px;
        white-space: pre-wrap;
        text-align: right;
        .h3();

        @media @mobileSM {
            .h5();
        }
    }
}
