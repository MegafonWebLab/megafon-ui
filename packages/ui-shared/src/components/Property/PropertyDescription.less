@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-beta-property-description');

@{b} {
    &__collapse {
        font-size: 12px;
        color: @link;
        text-decoration: underline dashed #34AAF2;
        cursor: pointer;
    }

    &__content-inner {
        margin-top: 8px;
    }
}
