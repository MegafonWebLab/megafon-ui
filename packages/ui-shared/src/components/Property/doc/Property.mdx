---
name: Property
menu: Конструктор
route: components/Property
---

import Property from '../Property';
import Example from './Property.example.mdx';
import Props from './Property.props.mdx';

# Property

<CopyToClipboardBox text="import { Property } from '@megafon/ui-shared';" />

<DoczTabs>
    <Example />
    <Props />
</DoczTabs>
