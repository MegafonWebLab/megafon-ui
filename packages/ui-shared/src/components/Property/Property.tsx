import React, { Ref } from 'react';
import PropTypes from 'prop-types';
import {Header, cnCreate, convert, TextLink} from '@megafon/ui-core';
import './Property.less';
import { Item } from './types';
import PropertyDescription from './PropertyDescription';

export interface IProperty {
    /** Ссылка на корневой элемент */
    rootRef?: Ref<HTMLDivElement>;
    /** Массив с данными для строки */
    items: Item[];
    /** Дополнительный класс для основного контейнера */
    className?: string;
    /** Текст для бейджа */
    badge?: string;
    /** Наличие нижней границы */
    borderBottom?: boolean;
    /** Единое значение для всей строки */
    mergedValue?: string;
    /** Иконка для строки */
    icon?: React.ReactNode;
    /** Несколько рядов в строке */
    multirow?: boolean;
}

const typographyConfig = {
    a: {
        component: TextLink,
        props: ['href', 'target'],
    },
};

const cn = cnCreate('mfui-beta-property');
const Property: React.FC<IProperty> = ({
    rootRef,
    items,
    className,
    badge = '',
    icon,
    borderBottom = false,
    mergedValue = '',
    multirow = false,
}) => {
    const renderTitle = React.useCallback(
        title =>
            title &&
            title.map((titleItem, i) => (
                <Header as={'h5'} key={i}>
                    {icon && i === 0 && (
                        <div className={cn('icon')}>{icon}</div>
                    )}
                    {convert(titleItem, typographyConfig)}
                </Header>
            )),
        []
    );

    const renderDescription = React.useCallback(
        description =>
            description &&
            description.map((descriptionItem, j) => (
                <div className={cn('desc')} key={j}>
                    <PropertyDescription {...descriptionItem} />
                </div>
            )),
        []
    );

    return (
        <div className={cn({ 'border-bottom': borderBottom }, [className])} ref={rootRef}>
            {badge && (
                <div className={cn('badge-wrapper')}>
                    <span className={cn('badge')}>{badge}</span>
                </div>
            )}
            <div className={cn('content')}>
                <div className={cn('items-wrapper')}>
                    {items.map(({ title, value, description }, i) => (
                        <div className={cn('item', { multirow })} key={i}>
                            <div className={cn('inner')}>
                                {renderTitle(title)}
                                {renderDescription(description)}
                            </div>
                            {
                                !mergedValue && (
                                    <div className={cn('value-wrapper')}>
                                        {value &&  (
                                            <span className={cn('value')}>{value}</span>
                                        )}
                                    </div>
                                )
                            }
                        </div>
                    ))}
                </div>
                {mergedValue && (
                    <div className={cn('value-wrapper', { merged: true })}>
                        <Header as="h3">{mergedValue}</Header>
                    </div>
                )}
            </div>
        </div>
    );
};

Property.propTypes = {
    rootRef: PropTypes.oneOfType([
        PropTypes.func,
        PropTypes.oneOfType([PropTypes.shape({ current: PropTypes.elementType }), PropTypes.any ]),
    ]),
    items: PropTypes.arrayOf(
        PropTypes.shape({
            title: PropTypes.arrayOf(PropTypes.string.isRequired),
            description: PropTypes.arrayOf(
                PropTypes.shape({
                    value: PropTypes.arrayOf(PropTypes.string).isRequired,
                    isCollapsible: PropTypes.bool,
                })
            ),
            value: PropTypes.string,
        }).isRequired
    ).isRequired,
    className: PropTypes.string,
    badge: PropTypes.string,
    borderBottom: PropTypes.bool,
    mergedValue: PropTypes.string,
    icon: PropTypes.node,
    multirow: PropTypes.bool,
};

export default Property;
