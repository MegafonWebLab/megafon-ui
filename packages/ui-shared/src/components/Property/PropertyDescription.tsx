import * as React from 'react';
import PropTypes from 'prop-types';
import {cnCreate, Collapse, Paragraph, convert, TextLink} from '@megafon/ui-core';
import './PropertyDescription.less';
import { Desc } from './types';

const typographyConfig = {
    a: {
        component: TextLink,
        props: ['href', 'target'],
    },
};

const cn = cnCreate('mfui-beta-property-description');
const PropertyDescription: React.FC<Desc> = ({
    value,
    isCollapsible = false,
}) => {
    const [isOpened, setIsOpened] = React.useState(false);

    const handleClickDesc = React.useCallback(() => setIsOpened(!isOpened), [
        isOpened,
    ]);

    const renderDescriptionItems = React.useCallback(
        () => (
            value.map((valueItem, i) => (
                <Paragraph hasMargin={false} key={i}>
                    {convert(valueItem, typographyConfig)}
                </Paragraph>
            ))
        ),
        [value]
    );

    if (isCollapsible) {
        return (
            <div className={cn()}>
                <span className={cn('collapse')} onClick={handleClickDesc}>
                    {isOpened ? 'Скрыть' : 'Подробнее'}
                </span>
                <Collapse
                    className={cn('content')}
                    classNameContainer={cn('content-inner')}
                    isOpened={isOpened}
                >
                    {renderDescriptionItems()}
                </Collapse>
            </div>
        );
    } else {
        return <div className={cn()}>{renderDescriptionItems()}</div>;
    }
};

PropertyDescription.propTypes = {
    value: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    isCollapsible: PropTypes.bool,
};

export default PropertyDescription;
