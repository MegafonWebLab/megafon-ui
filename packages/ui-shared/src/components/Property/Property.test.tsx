import * as React from 'react';
import { mount, shallow } from 'enzyme';
import WiFi from 'icons/Basic/32/Wi-fi_32.svg';
import Property from './Property';

describe('<Property />', () => {
    it('should render only title', () => {
        const wrapper = shallow(
            <Property
                items={[
                    { title: ['Звонки на все номера России'], value: '500 ₽' },
                ]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render title in several row', () => {
        const wrapper = shallow(
            <Property
                items={[
                    {
                        title: [
                            'Детализация в салонах МегаФона',
                            'Оплачивается каждый день заказанной детализации.',
                        ],
                        value: '2ГБ',
                    },
                ]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render title and description', () => {
        const wrapper = shallow(
            <Property
                items={[
                    {
                        title: ['Интернет на все сервисы'],
                        description: [
                            {
                                value: [
                                    'Boom, Yandex.Музыка, Zvooq, ВКонтакте Музыка',
                                ],
                            },
                        ],
                    },
                ]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render title and description in several rows', () => {
        const wrapper = shallow(
            <Property
                items={[
                    {
                        title: ['Интернет на все сервисы'],
                        description: [
                            {
                                value: [
                                    'Boom, Yandex.Музыка, Zvooq, ВКонтакте Музыка',
                                    'Google Play Music, Apple Music, онлайн радио.',
                                ],
                            },
                        ],
                        value: '500 ₽',
                    },
                ]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render title and hidden description', () => {
        const wrapper = shallow(
            <Property
                items={[
                    {
                        title: ['Музыкальные сервисы'],
                        description: [
                            {
                                value: [
                                    'Звонки на местные городские номера, когда вы находитесь дома,',
                                    'оплачиваются отдельно.',
                                ],
                                isCollapsible: true,
                            },
                        ],
                        value: '500 ₽',
                    },
                ]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with badge', () => {
        const wrapper = shallow(
            <Property
                badge={'Акция'}
                items={[
                    { title: ['Звонки на все номера России'], value: '500 ₽' },
                ]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with icon', () => {
        const wrapper = shallow(
            <Property
                icon={<WiFi style={{ display: 'block', fill: '#00B956' }} />}
                items={[{ title: ['Интернет на все сервисы'], value: '500 ₽' }]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with several descriptions', () => {
        const wrapper = shallow(
            <Property
                items={[
                    {
                        title: ['Доп звонки'],
                        value: '500Р',
                        description: [
                            {
                                value: [
                                    'После израсходования пакета минут звонки на номера МегаФона России предоставляются безлимитно.',
                                ],
                            },
                            {
                                value: [
                                    'Звонки на местные городские номера, когда вы находитесь дома, оплачиваются отдельно.',
                                ],
                                isCollapsible: true,
                            },
                        ],
                    },
                ]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with links', () => {
        const wrapper = shallow(
            <Property
                items={[
                    {
                        title: [
                            'Тарифы <a href="https://moscow.megafon.ru">Включайся!</a>, с которыми работает опция',
                        ],
                        value: '500 ₽',
                        description: [
                            {
                                value: [
                                    '<a href="https://moscow.megafon.ru">Выбирай, Смотри, Слушай, Говори, Общайся</a>',
                                ],
                            },
                        ],
                    },
                ]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render several rows', () => {
        const wrapper = shallow(
            <Property
                multirow
                items={[
                    { title: ['Абонентская плата:'] },
                    {
                        value: '500 ₽',
                        description: [
                            { value: ['Звонки на все номера России'] },
                        ],
                    },
                    {
                        value: '500 ₽',
                        description: [
                            { value: ['Звонки на все номера России'] },
                        ],
                    },
                    {
                        value: '500 ₽',
                        description: [
                            { value: ['Звонки на все номера России'] },
                        ],
                    },
                ]}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with border bottom', () => {
        const wrapper = shallow(
            <Property
                items={[
                    {
                        title: ['Звонки на все номера России'],
                        value: '500 ₽',
                    },
                ]}
                borderBottom
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should return reference to root element', () => {
        const ref: React.RefObject<HTMLDivElement> = React.createRef();
        mount(
            <Property
                items={[
                    { title: ['Звонки на все номера России'], value: '500 ₽' },
                ]}
                rootRef={ref}
            />
        );

        expect(ref.current).not.toBeNull();
    });
});
