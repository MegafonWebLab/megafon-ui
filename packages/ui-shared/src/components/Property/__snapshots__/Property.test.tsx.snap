// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Property /> should render only title 1`] = `
<div
  className="mfui-beta-property"
>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            Звонки на все номера России
          </Header>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500 ₽
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Property /> should render several rows 1`] = `
<div
  className="mfui-beta-property"
>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item mfui-beta-property__item_multirow"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            Абонентская плата:
          </Header>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        />
      </div>
      <div
        className="mfui-beta-property__item mfui-beta-property__item_multirow"
        key="1"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <div
            className="mfui-beta-property__desc"
            key="0"
          >
            <PropertyDescription
              value={
                Array [
                  "Звонки на все номера России",
                ]
              }
            />
          </div>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500 ₽
          </span>
        </div>
      </div>
      <div
        className="mfui-beta-property__item mfui-beta-property__item_multirow"
        key="2"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <div
            className="mfui-beta-property__desc"
            key="0"
          >
            <PropertyDescription
              value={
                Array [
                  "Звонки на все номера России",
                ]
              }
            />
          </div>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500 ₽
          </span>
        </div>
      </div>
      <div
        className="mfui-beta-property__item mfui-beta-property__item_multirow"
        key="3"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <div
            className="mfui-beta-property__desc"
            key="0"
          >
            <PropertyDescription
              value={
                Array [
                  "Звонки на все номера России",
                ]
              }
            />
          </div>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500 ₽
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Property /> should render title and description 1`] = `
<div
  className="mfui-beta-property"
>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            Интернет на все сервисы
          </Header>
          <div
            className="mfui-beta-property__desc"
            key="0"
          >
            <PropertyDescription
              value={
                Array [
                  "Boom, Yandex.Музыка, Zvooq, ВКонтакте Музыка",
                ]
              }
            />
          </div>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Property /> should render title and description in several rows 1`] = `
<div
  className="mfui-beta-property"
>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            Интернет на все сервисы
          </Header>
          <div
            className="mfui-beta-property__desc"
            key="0"
          >
            <PropertyDescription
              value={
                Array [
                  "Boom, Yandex.Музыка, Zvooq, ВКонтакте Музыка",
                  "Google Play Music, Apple Music, онлайн радио.",
                ]
              }
            />
          </div>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500 ₽
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Property /> should render title and hidden description 1`] = `
<div
  className="mfui-beta-property"
>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            Музыкальные сервисы
          </Header>
          <div
            className="mfui-beta-property__desc"
            key="0"
          >
            <PropertyDescription
              isCollapsible={true}
              value={
                Array [
                  "Звонки на местные городские номера, когда вы находитесь дома,",
                  "оплачиваются отдельно.",
                ]
              }
            />
          </div>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500 ₽
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Property /> should render title in several row 1`] = `
<div
  className="mfui-beta-property"
>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            Детализация в салонах МегаФона
          </Header>
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="1"
          >
            Оплачивается каждый день заказанной детализации.
          </Header>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            2ГБ
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Property /> should render with badge 1`] = `
<div
  className="mfui-beta-property"
>
  <div
    className="mfui-beta-property__badge-wrapper"
  >
    <span
      className="mfui-beta-property__badge"
    >
      Акция
    </span>
  </div>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            Звонки на все номера России
          </Header>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500 ₽
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Property /> should render with border bottom 1`] = `
<div
  className="mfui-beta-property mfui-beta-property_border-bottom"
>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            Звонки на все номера России
          </Header>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500 ₽
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Property /> should render with icon 1`] = `
<div
  className="mfui-beta-property"
>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            <div
              className="mfui-beta-property__icon"
            >
              <test-file-stub
                style={
                  Object {
                    "display": "block",
                    "fill": "#00B956",
                  }
                }
              />
            </div>
            Интернет на все сервисы
          </Header>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500 ₽
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Property /> should render with links 1`] = `
<div
  className="mfui-beta-property"
>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            Тарифы 
            <TextLink
              className=""
              color="blue"
              href="https://moscow.megafon.ru"
              key="1"
              underlineStyle="solid"
              underlineVisibility="hover"
            >
              Включайся!
            </TextLink>
            , с которыми работает опция
          </Header>
          <div
            className="mfui-beta-property__desc"
            key="0"
          >
            <PropertyDescription
              value={
                Array [
                  "<a href=\\"https://moscow.megafon.ru\\">Выбирай, Смотри, Слушай, Говори, Общайся</a>",
                ]
              }
            />
          </div>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500 ₽
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Property /> should render with several descriptions 1`] = `
<div
  className="mfui-beta-property"
>
  <div
    className="mfui-beta-property__content"
  >
    <div
      className="mfui-beta-property__items-wrapper"
    >
      <div
        className="mfui-beta-property__item"
        key="0"
      >
        <div
          className="mfui-beta-property__inner"
        >
          <Header
            as="h5"
            color="black"
            hAlign="inherit"
            key="0"
          >
            Доп звонки
          </Header>
          <div
            className="mfui-beta-property__desc"
            key="0"
          >
            <PropertyDescription
              value={
                Array [
                  "После израсходования пакета минут звонки на номера МегаФона России предоставляются безлимитно.",
                ]
              }
            />
          </div>
          <div
            className="mfui-beta-property__desc"
            key="1"
          >
            <PropertyDescription
              isCollapsible={true}
              value={
                Array [
                  "Звонки на местные городские номера, когда вы находитесь дома, оплачиваются отдельно.",
                ]
              }
            />
          </div>
        </div>
        <div
          className="mfui-beta-property__value-wrapper"
        >
          <span
            className="mfui-beta-property__value"
          >
            500Р
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
